cpp/src/utils/metrics/metric_cache.hh

```diff
@@ -0,0 +1,19 @@
+#ifndef TS_CPP_UTILS_METRICS_METRIC_CACHE_HH_
+#define TS_CPP_UTILS_METRICS_METRIC_CACHE_HH_
+
+#include <string>
+#include <map>
+
+#include "src/utils/metrics/metric.hh"
+
+namespace torchserve {
+    class MetricCache {
+        public:
+        virtual Metric AddMetric(const std::string& name, const std::string& unit,
```

Comment:
`LogMetricCache` implementation for backed which will derive from `MetricCache` will support the `AddMetric` and `GetMetric` API by:

1. Storing allowed metric names and corresponding unit and dimension names obtained from the metrics configuration.
2. Storing a map of type `std::map<std::string, LogMetric>`. Here, the key will be a string consisting of metric name, unit, dimension name and value corresponding to each dimension.

Reply:
are we going to have other implementations other than `LogMetricCache` in the future?

Reply:
Yes, we can have a `PrometheusMetricCache` which the frontend can use to store references to prometheus client metric objects.